### PR TITLE
convert fs-observable to typescript!

### PR DESCRIPTION
**Pull request by rgbkrk**
This is really great because we can remove some strange code and let typescript do it's thing thanks to typescript knowing node's `fs` well and `rxjs` having already been written in TypeScript. There's now less code and it's _more_ accurately typed.

### CODE REVIEW

**Comment by rgbkrk** [comment]
This one was strange enough for flow that I didn't create the `index.js.flow` file, instead opting to remove it from the flowconfig resolution.

**Comment by rgbkrk** [comment]
Now with some decent flow types!

I did have to `$FlowFixMe` one thing that I feel confident we'll be able to tackle as part of the typescript migration.

File: packages/fs-observable/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
+import * as fs from "fs";
```
Comment (rgbkrk) [inline packages/fs-observable/src/index.ts:1]: Could have sworn I did a `git mv index.js index.ts`. Perhaps my rebasing and the size of these changes caused `git` to say _nah_.

**Review by charlesjolley — APPROVED** [review]
LGTM!

File: packages/fs-observable/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+import { Observable, bindNodeCallback } from "rxjs";
+import { mergeMap } from "rxjs/operators";
+
+import mkdirp from "mkdirp";
+
+/**
+ * Deletes a file, if it exists
```
Comment (jdfreder) [inline packages/fs-observable/src/index.ts:9]: Hurray for 3rd person descriptive comments!

**Pull request merged**